cross_connect_client/models/cross_connect_server.py

```diff
@@ -0,0 +1,180 @@
+        headers["api-key"] = self.api_key
+        response = requests.request(
+            method,
+            self._absolute_url_for(url),
+            headers=headers,
+            json=data,
+            timeout=10,
+        )
+        response.raise_for_status()
+        return response.json()
+
+    def _get_cross_connect_url(self, redirect_url=None):
+        self.ensure_one()
+        groups = self.env.user.groups_id & self.group_ids
+        if not groups:
+            raise UserError(_("You are not allowed to access this server"))
+
+        if not self.env.user.email:
+            raise UserError(_("User email is required"))
+
+        data = {
+            "id": self.env.user.id,
+            "name": self.env.user.name,
+            "login": self.env.user.login,
+            "email": self.env.user.email,
+            "lang": self.env.user.lang,
+            "groups": [group.cross_connect_server_group_id for group in groups],
+        }
+        if redirect_url:
+            data["redirect_url"] = redirect_url
+
+        response = self._request("POST", "/access", data=data)
+        client_id = response.get("client_id")
+        token = response.get("token")
+        if not token:
+            raise UserError(_("Missing token"))
+
+        return self._absolute_url_for(f"login/{client_id}/{token}")
+
+    def _sync_groups(self):
+        self.ensure_one()
+        response = self._request("GET", "/sync")
+        remote_groups = response.get("groups", [])
+        # Removing groups that are not on the remote server
+        remote_groups_ids = {remote_group["id"] for remote_group in remote_groups}
+        self.group_ids.filtered(
+            lambda group: group.cross_connect_server_group_id not in remote_groups_ids
+        ).unlink()
+
+        # Create or Update existing groups
+        for remote_group in remote_groups:
+            existing_group = self.group_ids.filtered(
+                lambda group: group.cross_connect_server_group_id == remote_group["id"]
+            )
+            if existing_group:
+                existing_group.sudo().write(
+                    {
+                        "name": f"{self.name}: {remote_group['name']}",
+                        "comment": remote_group["comment"],
+                    }
+                )
+            else:
+                self.env["res.groups"].sudo().create(
+                    {
+                        "cross_connect_server_id": self.id,
+                        "cross_connect_server_group_id": remote_group["id"],
+                        "name": f"{self.name}: {remote_group['name']}",
+                        "comment": remote_group["comment"],
+                    }
+                )
+
+    def action_sync(self):
+        for record in self:
+            record._sync_groups()
+
+    def action_disable(self):
+        for record in self:
+            record.group_ids.unlink()
+
+    @property
+    def _server_env_fields(self):
+        return {"api_key": {}}
```

Comment:
I think we should override the `unlink` method to delete the groups and menu before deleting the cross connect server.
(Even without changing the `action_disable` method, I guess the `cross.connect.server` could be deleted while enabled)

Reply:
Can’t we cascade these?

Reply:
Better indeed.